Add various Samtec mezzanine footprints from the Q Strip/Q Pairs, Basic Blade & Beam, Razor Beam, and mPOWER families

calebreister opened the pull request:
This pull request contains footprints for [Altera HSMC cards](https://www.intel.com/content/dam/www/programmable/us/en/pdfs/literature/ds/hsmc_spec.pdf). There are two variations, which use different connectors:

1. [Samtec ASP-122952-01](http://suddendocs.samtec.com/prints/asp-122952-01-mkt.pdf): a modified Samtec QTH-series connector with every third pin removed in Bank 1 to improve signal integrity for transceivers.
2. [Samtec QTH-090-01-x-D-A](https://www.samtec.com/products/qth-090-01-l-d-a-tr): identical to the ASP-122952-01, except that all of the pins were left in Bank 1. It is not as good for transceivers, but is much more common.

The following footprints were added...

Samtec_HSMC_ASP-122952-01_P0.5mm_Vertical
![screenshot_20190115_182233](https://user-images.githubusercontent.com/6203200/51222434-8a1fc180-18f2-11e9-89ec-f543072c2641.png)

Samtec_QTH-090-01-x-D-A_P0.5mm_Vertical
![screenshot_20190115_182042](https://user-images.githubusercontent.com/6203200/51222368-488f1680-18f2-11e9-8260-93e27e508846.png)

Samtec_HSMC_Card_QTH-090-01-x-D-A_3.076x0.932in_P0.5mm_Vertical
![screenshot_20190115_181745](https://user-images.githubusercontent.com/6203200/51222330-26959400-18f2-11e9-9307-c6d804cd7929.png)

**KLC Notes:**
- The `HSMC_Card` footprint anchors are centered on the connector, not the courtyard. Changing this would cause issues with the 3D models.
- The `HSMC_Card` courtyards do not align to a 0.01mm grid, since the specifications were provided by the HSMC specification.
- Some 3D models are reused, so the footprint names do not always match.

**See also:**
- [Pull request](https://github.com/KiCad/kicad-symbols/pull/1428) for `HSMC_Card` symbols
- [Pull request](https://github.com/KiCad/kicad-packages3D/pull/512) for the 3D Models

------------

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [x] Provide a URL to a datasheet for the footprint(s) you are contributing
- [x] An example screenshot image is very helpful 
- [x] If there are matching symbol or 3D model pull requests, provide link(s) as appropriate
- [x] Check the output of the Travis automated check scripts - fix any errors as required

Commit:
Add HSMC card connector footprints.

Commit:
Add HSMC host footprints

Commit:
Modify QTH/QSH pin numbers to match actual connector (not the HSMC pinout).

Commit:
Renumber pins and other minor modifications to HSMC footprints.

Commit:
Make base versions of QTH-090-01-x-D-A and ASP-122952 comply with KLC

Commit:
Create KLC-compliant HSMC card footprints, remove HSMC host footprints

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KiCad/kicad-footprints?pullRequest=1304) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KiCad/kicad-footprints?pullRequest=1304) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KiCad/kicad-footprints?pullRequest=1304) it.</sub>

calebreister commented:
A licensing issue has come up with the 3D models, so a minor revision may be in order.

poeschlr commented:
A footprint pull request is never dependent on contributing a 3d model. All that needs to be done is preparing the footprint to accept a 3d model. (Meaning filling out the 3d model settings such that it is easy to add a model at any time without needing to edit the footprint.)

evanshultz commented:
Why is the courtyard not on a 0.01mm grid? Can you provide that reference? I don't see anything in the drawings linked above.

The QTH link above is for the product page. Would http://suddendocs.samtec.com/prints/qth-xxx-xx-x-d-xxx-mkt.pdf (or possibly http://suddendocs.samtec.com/prints/qth-xxx-xx-x-d-xxx-footprint.pdf) be a better link to use?

evanshultz commented:
Mark the PCB edge explicit, then. This does not affect the courtyard which should conform to normal KLC requirements. See https://github.com/KiCad/kicad-footprints/pull/570 and several footprints in the library (search for "edge" in the footprint name). This is not in KLC so I understand it's not obvious what to do.

In combination don't those two docs cover everything? What is missing when both are included?

calebreister commented:
I see. Maybe something about PCB edges should be added to the KLC. Since the HSMC specification sets the width of the board, should I put a PCB edge label on three sides?

Those documents do cover everything, but the footprint description only allows a one URL.

I will revise the `HSMC_Card` footprints with the PCB edges denoted in Dwgs.User. Do I need to make a new pull request, or is there a way to make the change in this one?

calebreister commented:
Should the mounting holes used in the footprint also be added to the footprint library?

poeschlr commented:
What would be the reason for the need to define board edges. If this is a pure connector footprint (Meaning it fits into the connector lib) then there should not be any board edge defined.

If this is for including a full module on top (or below) of ones pcb then the board edges of that subsystem do not influence the board edges of your own pcb. The board edges of such a subsystem would then be on the fab layer and the footprint would live in one of the module libraries.

---

The only reason i can think of why there is a need for a pcb edge defined is if there is anything in the way. (This would be the case if this is designed to fit into a specific housing that is part of the system for which we design the interface definition here.)
Or if this is intended to define the board shape for creating a clone of the system in question.
We would need to think about which library such a thing should be placed in.
It might even be better to create this as a template instead of a single footprint (one would then get the correct position file entry for the connector. And BOM entries for the mounting holes.)

calebreister commented:
HSMC is a standard developed by Altera (now Intel FPGA) for expansion cards that attach to FPGA development boards. After realizing that templates exist (I didn't until after developing the footprints), I agree that only the connectors themselves should be placed in the footprint library. Thus, I plan to port the `HSMC_Card` footprints over to a template and possibly perform some minor revisions to the connector footprints.

evanshultz commented:
Please let me know when you've made the changes and are ready for a review then. Thanks!

Commit:
Remove HSMC_Card footprints (they will be implemented as templates). Minor
edits to ASP-122952-01 and QTH-090-x-D-A connector footprints.

calebreister commented:
As suggested, I am going to migrate the HSMC_Card footprints to templates, and have thus removed them completely. A few minor edits were also made:
- Local clearance settings for the NPTH mounting holes were set to 0
- Pin outlines were removed from F.Fab.

Commit:
Add 0.125in mounting hole (a common imperial hole size), used by the HSMC
card template.

Commit:
Make MountingHole_0.125in KLC-compliant

Commit:
Merge remote-tracking branch 'upstream/master' into CalebReister

evanshultz commented:
@calebreister 
So this will only be the connector footprints. Just to clarify in case I missed something.

Mounting hole:
1. According to http://kicad-pcb.org/libraries/klc/G1.9/, metric units are preferred. Would it be possible to change the mounting hole to 3.175mm (you can use inches in the description for searchability)?

Connectors:
1. It looks like each pad has a clearance set. Why is this? If it's needed, can you do it per-footprint instead of per-pad?
1. We typically put the silk ref des above the part.
1. Silk offset from the fab lines isn't well-defined, but >0.4mm is quite a bit. The pin 1 mark will be more clear if the silk is tighter as well. Could you scoot in the silk lines?
1. They are slightly shifted. Is this intentional? For example, pin 103 on the QTH footprint has an X location of 3.25mm and the ASP footprint's corresponding pin 83 is at 3.24866mm. And on the ASP footprint pins 2-4 are 197mil apart but pins 6-8 are 196mil. The numbers are incredibly precise but not quite accurate which makes me think of a rounding or unit conversion issue.
1. The fab pin 1 mark doesn't quite line up perfectly with the horizontal fab layer line. In addition, it could be a bit bigger.
1. Many of the dimensions are not quite to http://suddendocs.samtec.com/prints/asp-122952-01-mkt.pdf. Can you please take a look and let me know if there's a better doc to use or what is going on?
![image](https://user-images.githubusercontent.com/1936989/59802411-e15d0000-929d-11e9-9fde-1052455ec375.png)

Lastly, are there additional footprint variants? Not that you need to add them, but http://suddendocs.samtec.com/catalog_english/qth.pdf indicates more than just the -090 variant may be available. And https://www.intel.com/content/dam/www/programmable/us/en/pdfs/literature/ds/hsmc_spec.pdf references ASP-122953-01. So I wanted to check if you knew about them.

calebreister commented:
@evanshultz 

Sorry it took so long for me to respond. Yes, this PR contains the ASP-122952-01 and QTH-090-01-L-D-A connector footprints, and possibly one mounting hole. Many of the issues you mentioned are likely errors, and I'll try to fix them when I get the chance.

I am aware that other variants of the same connector exist. The ASP-122953-01 is the mating connector for the '122952 that goes on host boards that would likely need to be remade from scratch. I've considered trying to make a script to generate QTH connector footprints (such as camera modules on the NVIDIA Jetson boards), but so far haven't had the time. If I were to create such a script, is there a preferred language?

evanshultz commented:
Thanks for responding! When you can, let us know what you find and if I've gotten anything wrong.

Scripts for footprints are at https://github.com/pointhi/kicad-footprint-generator, and for yours specifically at https://github.com/pointhi/kicad-footprint-generator/tree/master/scripts/Connector/Connector_Samtec. It's all in Python.

calebreister commented:
Update: I've begun work on a footprint wizard for the vertical variants of the [Samtec Q Strip family](https://www.samtec.com/connectors/high-speed-board-to-board/mezzanine/qstrip) (QSH/QTH, QSS/QTS, and QSE/QTE). At the moment, the code is a little messy, but I'm working on cleaning it up.

https://github.com/calebreister/kicad-footprint-wizards/blob/CalebReister/QStrip_wizard.py

Commit:
Create new wizard-generated Samtec_HSMC_ASP122952-01 and Samtec_QTH-090-01-x-D-A footprints

Commit:
Fix minor KLC issues for Samtec_QTH-090-01-x-D-A and Samtec_HSMC_ASP-122952-01

calebreister commented:
@evanshultz 

I've replaced the QTH-090-01-x-D-A and ASP-122952-01 footprints with wizard-generated versions. Considering the number of possible variations of these connectors (likely several hundred), it may be worth creating a `Connector_Samtec_QStrip` library.

Unless I can find a way to automate the wizard, I may try porting is over to the KicadModTree library you initially recommended, since a YAML file can be used to generate multiple footprints. Another option may be to write a Python script that instantiates the wizard, sets parameters, and generates the footprints.

Commit:
Regenerate Samtec_HSMC_ASP-122952-01_P0.5mm_Vertical using new script

calebreister commented:
I rewrote the script using the KicadModTree, and was able to generate 74 Q Strip footprints succesfully (including the ASP-122952-01 and ASP-122953-01). I plan on submitting a pull request for my script (and YAML file) to [kicad-footprint-generator](https://github.com/pointhi/kicad-footprint-generator). Should add the additional footprints to this PR or create a separate one?

evanshultz commented:
I'm fine to just add them here. We won't have to review each individual one since they are generated by a script, so it doesn't actually increase the review burden to pile them on this PR.

Commit:
Add new script-generated Samtec Q Strip and Q Pairs connectors

Commit:
Rename MountingHole_0.125in to MountingHole_3.175mm

calebreister commented:
@evanshultz You may want to check the metadata, I may have over-populated it.

Commit:
Add Connector_Samtec_QStrip_QPairs entry to fp-lib-table

Commit:
Merge tag '5.1.3' into CalebReister

Footprints released in conjunction with KiCad 5.1.3

Commit:
Merge tag '5.1.5' into CalebReister

Footprints released in conjunction with KiCad 5.1.5

Commit:
Fix duplicated entries in fp-lib-table

Commit:
Fix another duplicate entry in fp-lib-table

Commit:
Merge tag '5.1.6' into CalebReister

Footprints released in conjunction with KiCad 5.1.6

calebreister commented:
I have started work on creating the right-angle variations of the Q Strip/Q Pairs footprints. I'm also taking a look at creating the Razor Beam footprints. A few LSHM (0.5mm) connectors already exist in Connector_Samtec, but my script should be able to generate a large portion of the Razor Beam family (LSHM, LSS, and LSEM).

Commit:
Change plane pin naming convention in QStrip_QPairs connectors

Commit:
Add auto-generated Samtec_RazorBeam footprints (LSEM and LSHM series)

Commit:
Fix 3D model paths in QStrip_QPairs and RazorBeam footprints

calebreister commented:
@evanshultz I've made considerable improvements to both the scripts and footprints since we last talked nearly a year ago. I've even added a number of connectors from the RazorBeam family, and may also try to create some footprints for the right-angle variants. Any updates on when/if these connectors can be merged?

calebreister commented:
Closing for the moment while I work on cleaning up the code and adding even more footprints.

Commit:
+ Redraw QStrip/QPairs sockets with Pin 1 in upper left
+ Fix pin numbering error on QStrip/QPairs connectors

calebreister commented:
Some of the footprint specifications have the following note on certain plated through holes:
> TO BE PROCESSED USING PASTE-IN-HOLE TECHNOLOGY

Should these footprints include paste over the holes designated in the datasheet(s)?

Example: [QTH-RA](http://suddendocs.samtec.com/prints/qth-xxx-xx-xxx-d-ra-xxx-footprint.pdf)

calebreister commented:
With minor modifications, I can get my scripts to also generate the Basic Blade & Beam Strip (such as BSH/BTH) connector families as well.

calebreister commented:
Since this PR seems to keep widening in scope, I'm going to close it for the time being so I can give the scripts a bit more work. Then I will make a new request.

Commit:
Rename Connector_Samtec_QStrip_QPairs library to Connector_Samtec_QSeries

Commit:
QStrip_QPairs_Vertical script update, minor metadata changes

Commit:
+ Connector_Samtec_QStrip: Correct hole position errors in Q Strip/Q Pairs sockets
+ Add Samtec BasicBladeAndBeam library with BTH and BSH connectors

calebreister commented:
@myfreescalewebpage This PR is not abandoned. I am continuing work on it.

Commit:
Connector_Samtec_BasicBladeAndBeam:
+ Corrections to BSH/BTH connectors
+ Add BTS connectors

Commit:
Metadata update in Connector_Samtec, Connector_Samtec_QStrip, Connector_Samtec_RazorBeam,
and Connector_Samtec_BasicBladeAndBeam

Commit:
Connector_Samtec_BasicBladeAndBeam
+ Add BSH connectors
+ Correct dimensions in all families

Commit:
Fix tag error in Samtec_HSMC_ASP-122953-01_P0.5mm_Vertical

myfreescalewebpage commented:
Sorry!

Commit:
Regenerate Samtec_QStrip vertical connectors

Commit:
Regenerate Connector_Samtec_BasicBladeAndBeam footprints
+ Correct file names (BTS/BSS and BTE/BSE had the wrong pitch)
+ Various minor adjustments/corrections

Commit:
Remove some in-progress footprints to prepare for library migration
- Horizontal connectors in Connector_Samtec_QStrip
- Connector_Samtec_RazorBeam library

calebreister commented:
In preparation for the library migration and the upcoming KiCad 6 update, I've narrowed the scope of this PR to the connectors whose scripts I have already completed. This includes the vertical variants of the following families: QTH/QSH, QTS/QSS, QTE/QSE, BTH/BSH, BTS/BSS, BTE/BSE.

calebreister commented:
Here is a sample of a few of the connectors added.

![Screenshot_20200907_204917](https://user-images.githubusercontent.com/6203200/92431170-a9183a80-f14b-11ea-9ca5-cabc05966c84.png)

![Screenshot_20200907_205252](https://user-images.githubusercontent.com/6203200/92431332-1af08400-f14c-11ea-9a44-f39ff8dfb77f.png)

![Screenshot_20200907_205604](https://user-images.githubusercontent.com/6203200/92431512-8fc3be00-f14c-11ea-9675-840f495ab457.png)

Commit:
Minor corrections in Connector_Samtec_QStrip

chschlue commented:
As long as you have a GitLab account as described in the README, you can just keep working on this PR as before.
Just saying, no need to narrow it down.

calebreister commented:
@chschlue I do have a GitLab account with the same email as this account, both are public. The reason I've narrowed the scope is so that I can use the GitLab repo instead of this one once it is available.

Commit:
Minor corrections in Connector_Samtec_QStrip library

justyn committed:
Fix pin numbering in 1.5mm pitch Pico-Lock connectors (504050 series) (#2466)

The existing footprints have reversed pin ordering, so that pad 1 does
not match that on the datasheet or indicated by the pin 1 indicator on
the connector itself.

The footprints are created with kicad-footprint-generator with

```
smd_single_row_plus_mounting_pad.py conn_molex.yaml
```

Footprints generated by this script must always have pin 1 on the
left-most side, so in order to fix the numbering the connectors are
rotated by 180 degrees.

This update also fixes incorrect tags that specified "top entry" instead
of "horizontal".

(cherry picked from commit 87502278c177d8aa220606418734544c2e591dd9)

Commit:
Correct tags for Samtec HSMC connectors, regenerate Q Strip/Q Pairs
vertical connectors

Commit:
Add Connector_Samtec_RazorBeam.pretty, correct datasheet locations
in Connector_Samtec_QStrip.pretty so that the links work correctly

calebreister commented:
I've created a new script capable of generating all of the vertical Razor Beam connectors (LSHM, LSS, LSEM). This includes the LSHM connectors, a few of which are already included in the Connector_Samtec library. Shall I delete the old versions of these connectors from this branch since newer versions are now available in Connector_Samtec_RazorBeam?

calebreister commented:
Razor Beam connectors:
![Screenshot_20200920_124140](https://user-images.githubusercontent.com/6203200/93720557-aef41f80-fb3e-11ea-8c1d-97f5f5a0c2d4.png)

Samtec 3D model fit test:
![Screenshot_20200920_124309](https://user-images.githubusercontent.com/6203200/93720587-db0fa080-fb3e-11ea-80c8-e2de737582ef.png)

calebreister commented:
There is one minor instance of clipping: the alignment holes for the LSEM connectors are supposed to be centered at -0.95mm according to the drawing and datasheet, but the alignment pins on the model are closer to -0.1mm. I think that the error is most likely in the model. Even if it isn't, there's enough clearance in the holes and enough space on the pads for the connector to shift 0.05mm.

Commit:
Add Samtec mPOWER (UMPT/UMPS) connector footprints

Commit:
Add "MP" pin names to Samtec mPOWER connectors with weld tabs

calebreister commented:
Added the Samtec mPOWER connector family (UMPT/UMPS) to Connector_Samtec.pretty.

Here are a couple of the new footprints. These are the weld tab variants.
![Screenshot_20200920_195142](https://user-images.githubusercontent.com/6203200/93729788-d1a42980-fb7a-11ea-86fd-04f8d68d589f.png)

They match the Samtec 3D models.
![Screenshot_20200920_195629](https://user-images.githubusercontent.com/6203200/93729956-63ac3200-fb7b-11ea-8687-7299598b393f.png)

saviocmc committed:
Add OnSemi QRE1113 and Everlight ITR1201SR10AR (#2482)

* Add OnSemi CASE 100AQ, for the QRE1113 sensor

This is the Through Hole footprint for the QRE1113
Miniature Reflective Object Sensor, a reflective optocouple

Reference:
https://www.onsemi.com/pub/Collateral/QRE1113-D.PDF

This footprint is required to fix issue 2623 on kicad-symbols
https://github.com/KiCad/kicad-symbols/issues/2623

* Add footprint for Everlight ITR1201SR10AR sensor

This is the SMD footprint for the Everlight ITR1201SR10AR
Miniature Reflective Object Sensor, a reflective optocouple

Reference:
https://www.everlight.com/file/ProductFile/ITR1201SR10AR-TR.pdf

This footprint is required to fix issue 2623 on kicad-symbols
https://github.com/KiCad/kicad-symbols/issues/2623

* Fix 3D model file path for Everlight_ITR1201SR10AR and OnSemi_CASE100AQ

The fix was in order to make them comply with the KiCad Library Convention F9.3

* Fix OnSemi_CASE100AQ Pad 1 shape, setting it as a Rounded Rectangle

The change was made in compliance with the KiCad Library Convention F7.3

* Fix OnSemi_CASE100AQ positioning, setting the origin on Pad 1

The change was made in compliance with the KiCad Library Convention F7.2

(cherry picked from commit 78041bd683b5285bf76878afdd5e1394f7b034c9)

evanshultz committed:
Use unique and IPC package sizes for R, L, and C (#2280)

* Use unique and IPC package sizes for R, L, and C

(cherry picked from commit 0008170b644c3ba77b1f9f4eafd7a40fd624de7a)

cpresser committed:
Fix L_1806 by reverting to previous version (#2487)

(cherry picked from commit 806e461bc6492ab889ea86f47da6980d241283d2)

Commit:
Merge tag '5.1.7' into CalebReister

Commit:
Rename footprints in Connector_Samtec_RazorBeam to comply with KLC

Commit:
Rename Samtec Basic Blade and Beam connectors for KLC compliance

Commit:
Rename Samtec mPOWER (UMPT/UMPS) connectors to meet KLC guidelines

Commit:
Delete old Samtec LSHM connector footprints

calebreister commented:
Since I now have a script that generates all of the Samtec Razor Beam (LSHM/LSS/LSEM) footprints, I have deleted the old LSHM footprints in Connector_Samtec.pretty.

evanshultz commented:
@calebreister 
Back from a long hiatus on this! Though I never assigned it to myself.

Anyway, let's try to get this done. My thought is to review the script and then spot-check some footprints. If that looks good, we pull the trigger on the whole lot of them. If you're willing to handle any script updates if there are problems, so far as you can see (we can't know the future), that should be fine.

Let me start with a couple things:
1. Why have non-Samtec footprints been touched? SMT caps/fuses/diodes/LEDs/resistors (stuff from the same 2-terminal generator script), Molex connectors, etc. Is any of that intentional? If so, why?
1. Can all of these footprints just go into Connector_Samtec? Do they need their own folder? Maybe a breakdown of the quantities of footprints added in each new folder would help?

calebreister commented:
@evanshultz 

> Why have non-Samtec footprints been touched? SMT caps/fuses/diodes/LEDs/resistors (stuff from the same 2-terminal generator script), Molex connectors, etc. Is any of that intentional? If so, why?

Not intentional in the slightest. I tried to merge in changes from each release tag, but may have screwed it up. I'm not sure how to fix that.

> Can all of these footprints just go into Connector_Samtec? Do they need their own folder? Maybe a breakdown of the quantities of footprints added in each new folder would help?

It's quite a few connectors, with room for expansion later on. Here is the breakdown:
- Connector_Samtec_QStrip.pretty (72 footprints): Q Strip and Q Pairs (QTH/QSH, QTS/QSS, and QTE/QSE) connectors
- Connector_Samtec_BasicBladeAndBeam.pretty (64 footprints): Basic Blade & Beam (BTH/BSH, BTS/BSS, BTE/BSE) connectors. These have a similar layout to Q Strip connectors (they are in fact generated using the same script), but do not have a ground strip.
- Connector_Samtec_RazorBeam.pretty (22 footprints): Razor Beam self-mating connectors (LSHM, LSS, and LSEM)
- Connector_Samtec.pretty (add 18, remove 12 footprints): added a couple of application-specific connector footprints (Q Strip connectors with the first bank differential) and vertical mPOWER (UMPT/UMPS) power connectors, which are designed to stack with a variety of other Samtec connectors (including Q Strip/Q Pairs and Basic Blade & Beam). Removed old LSHM footprints, replacements in Connector_Samtec_RazorBeam.pretty.

I created these libraries due to the number of variants and the possibility for future expansion. I've already begun work on making the horizontal variants of many of these footprints. This will cause some of the libraries to nearly double in size. I'm also working on creating a library for the [Q2](https://www.samtec.com/?Q2) family of connectors. That will most likely need to be another PR in the (possibly not-so-near) future.

I will be happy to oblige with script updates. I've put a lot of work into these scripts and the associated JSON files containing the footprint specifications. I also matched a sizeable sample of the footprints up to 3D models from the Samtec website.

calebreister commented:
Note that the current script PR has some incomplete/work-in-progress scripts (such as QStrip_Vertical.py) that I haven't touched in awhile. Since I've done some major restructuring, I will likely end up deleting those files and rewriting them in a new branch.

calebreister commented:
Quick question about the Q Strip/Q Pairs connectors. What should the ground plane pads be called? The planes are independent per bank, but all of the pads within a bank are connected. Currently, they are prefixed with a "P" (for plane) and the bank number (see above). I thought about naming them SH, but they are not *technically* shielding. What should they be called?

calebreister commented:
@evanshultz What do I need to do to resolve those conflicts? Can I just delete the files and merge in the upstream master branch?

Commit:
Merge branch 'master' into CalebReister
